package-builder/flox-build.mk

```
@@ -490,15 +491,27 @@ define BUILD_nix_sandbox_template =
   # build cache.  This is used for (at least) publish.
   $(eval _do_buildCache = $(if $(DISABLE_BUILDCACHE),,true))
 
+  # 'git ls-files' will list all _tracked_ files **including deleted files**.
+  # Consequently, when we try to create a tarball with all files listed by
+  # 'git ls-files' we may attempt packing files that are actually deleted.
+  # To avoid this, we filter out deleted files.
+  # Because 'git ls-files' does not have a flag to filter deleted files,
+  # but allows to _only_ show deleted files, use 'comm' to do the filtering for
+  # us.
+  # Assumes the output of 'git ls-files' to be sorted.
+  $(eval $(_pvarname)_src_list = $($(_pvarname)_tmpBasename)/src-list)
+  $($(_pvarname)_src_list):
+	$(_comm) -23 <($(_git) ls-files -c) <($(_git) ls-files -d) > $$@
+
   # The sourceTarball value needs to be stable when nothing changes across
   # builds, so we create a tarball at a stable temporary path and pass that
   # to the derivation instead.
   $(eval $(_pvarname)_src_tar = $($(_pvarname)_tmpBasename)/src.tar)
-  $($(_pvarname)_src_tar): $(PROJECT_TMPDIR)/check-build-prerequisites
+  $($(_pvarname)_src_tar): $($(_pvarname)_src_list) $(PROJECT_TMPDIR)/check-build-prerequisites
```

```suggestion
  $($(_pvarname)_src_tar): $($(_pvarname)_src_list)
```

Moved `$(PROJECT_TMPDIR)/check-build-prerequisites` to be a prerequisite of `$($(_pvarname)_src_list)` - nicer to have a linear chain of dependencies in the build graph where possible.